Settings.c

```diff
@@ -605,7 +605,7 @@ static void writeMeterModes(const Settings* this, FILE* fd, char separator, unsi
    if (this->hColumns[column].len) {
       const char* sep = "";
       for (size_t i = 0; i < this->hColumns[column].len; i++) {
-         fprintf(fd, "%s%d", sep, this->hColumns[column].modes[i]);
+         fprintf(fd, "%s%d", sep, (int)this->hColumns[column].modes[i]);
```

Comment:
Avoid this cast by having `MeterModeId` be an alias for `int`; which is also basically what the enum would have been; `uint8_t` doesn't save anything, as inside the structs where it's used it causes padding to be used thus negating any gains …

Reply:
@BenBE 
Um... This use of `MeterModeId` is an array in `MeterColumnSetting`. Not part of `Meter` structures.

Reply:
Sure. I was referring to the use as part of `struct Meter_`.

Also, the format string is `%d` in this line; thus forcing your cast.

Reply:
I don't know. Is the `MeterModeId` array in `MeterColumnSetting` meant to have a short lifetime? If that's the case, then allocating a bigger `int` array wouldn't hurt.

(Note that in a variadic function like `printf`, any argument shorter than `int` would implicitly promote to `int` anyway even without the cast.)